changelog.d/15773.feature

```diff
@@ -0,0 +1 @@
+Allow configuring the set of workers to proxy outbound federation traffic through via `outbound_federation_restricted_to`.
```

Comment:
> Otherwise looks good to me

@erikjohnston Are we fine with merging even if the base is mostly your code here?